### PR TITLE
docs: Add hint about performance impact of VM snapshots

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

File: docs/WritingTests.asciidoc [hunk 1]
```diff
@@ -261,6 +261,19 @@ the clone source job or specifying the variable explicitly:
 clone_job.pl --from https://openqa.opensuse.org  --host localhost 24 TESTDEBUG=1 SKIPTO=consoletest-yast2_i
 ----
 
+==== Performance impact
+
+Keep in mind that creating VM snapshots can induce quite high I/O load. This
+should not impact the stability of openQA jobs but can increase job execution
+time. If you run jobs on a machine where responsiveness of other services
+matter, for example your desktop machine, consider executing the worker
+services with +ionice+ to reduce the risk of your system becoming
```
Comment (coolo) [inline docs/WritingTests.asciidoc]: No, don't do that - patch the systemd service instead:

https://www.freedesktop.org/software/systemd/man/systemd.directives.html#IOSchedulingPriority=

File: docs/WritingTests.asciidoc [hunk 2]
```diff
@@ -261,6 +261,19 @@ the clone source job or specifying the variable explicitly:
 clone_job.pl --from https://openqa.opensuse.org  --host localhost 24 TESTDEBUG=1 SKIPTO=consoletest-yast2_i
 ----
 
+==== Performance impact
```
Comment (coolo) [inline docs/WritingTests.asciidoc]: I find the headline a bit strange. We are here in Writing Tests documentation - and then you throw a performance impact into it. I would expect the paragraph you wrote in Pitfalls actually

File: docs/WritingTests.asciidoc (hunk 1 above)
Comment (okurz) [reply docs/WritingTests.asciidoc]: fixed

File: docs/WritingTests.asciidoc (hunk 2 above)
Comment (okurz) [reply docs/WritingTests.asciidoc]: done, updated. please review again

File: docs/Pitfalls.asciidoc [hunk 3]
```diff
@@ -36,3 +36,20 @@ user can share the environment without problems.
...
+== Performance impact
+
+openQA workers can cause high I/O load, especially when creating VM snapshots.
+The impact therefore gets more severe when +MAKETESTSNAPSHOTS+ is enabled.
+should not impact the stability of openQA jobs but can increase job execution
+time. If you run jobs on a machine where responsiveness of other services
+matter, for example your desktop machine, consider patching the
+IOSchedulingPriority of a workers service file as described in the
+https://www.freedesktop.org/software/systemd/man/systemd.exec.html#IOSchedulingPriority=[systemd
+documentation], for example set +IOSchedulingPriority=7+ for the best-effort
```
Comment (coolo) [inline docs/Pitfalls.asciidoc]: If you want best-effort you need to set IOSchedulingClass=best-effort and not the priority

Comment (okurz) [reply docs/Pitfalls.asciidoc]: isn't best-effort the default scheduling class? and if it is then my intention was to set the priority only within the default class

Comment (coolo) [reply docs/Pitfalls.asciidoc]: then just remove the mentioning of the IO scheduling class?

Comment (okurz) [reply docs/Pitfalls.asciidoc]: yes, sir!

File: docs/Pitfalls.asciidoc [hunk 4]
```diff
@@ -36,3 +36,20 @@ user can share the environment without problems.
 This approach will lead to a problem when the openqa package is updated, since the
 directory permissions will be changed again, nothing a `chmod -R g+rwx /var/lib/openqa/`
 and `chgrp -R openqa /var/lib/openqa` can not fix.
+
+== Performance impact
+
+openQA workers can cause high I/O load, especially when creating VM snapshots.
+The impact therefore gets more severe when +MAKETESTSNAPSHOTS+ is enabled.
```
Comment (coolo) [inline docs/Pitfalls.asciidoc]: As I'm looking at the rendered version:

+ is not changing the rendering at all - asciidoc quick reference has also no mentioning of this. I guess we want to use `` as monospace for these variables? (unrelated to this PR actually, we markup openqa variables inconsistently all over)

Comment (okurz) [reply docs/Pitfalls.asciidoc]: It works for me when I transform the .asciidoc into .html with `asciidoc`. Also http://www.methods.co.nz/asciidoc/userguide.html#_text_formatting mentions it for monospace so … wut?

Comment (coolo) [reply docs/Pitfalls.asciidoc]: http://open.qa/docs/#_networking_in_openqa is the only place where I see monospace - and it's using backticks. And for reference I checked http://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#formatted-text

Comment (coolo) [reply docs/Pitfalls.asciidoc]: https://github.com/powerman/asciidoc-cheatsheet/blob/master/README.adoc also talks about +mono+ and works on github while our ++s don't. very strange

Comment (coolo) [reply docs/Pitfalls.asciidoc]: Found it! http://asciidoctor.org/news/2014/08/12/asciidoctor-1-5-0-released/#means-passthrough - your document was updated last 2013 - 2014 changed the meaning of + to not affect rendering

Comment (okurz) [reply docs/Pitfalls.asciidoc]: Well, I have a change prepared replacing all '+' with '`' now but I think the use of `+` is safer so I like to keep that.

Comment (coolo) [reply docs/Pitfalls.asciidoc]: well, you might want + *and ` meanwhile.

**Pull request merged**